docs/demo.md

```diff
@@ -0,0 +1,218 @@
+  Sometimes the best choice is to leave an arg undefined.
+
+#### Story `argTypes`
+
+Standard convention for `argTypes` exists across dozens of existing stories.
+Study demo code for details. Most importantly, main component props that are
+_not_ auto-generated must be properly categorized. The two acceptable story
+categories are:
+
+- Subcomponent. All subcomponent props (including `children`) are specified
+  under the associated subcomponent name.
+- "Story". This category includes all ancillary controls that are needed to
+  stress the full flexibility of the rendered component(s).
+
+#### Story `parameters`
+
+Currently, story `parameters` are used to link to the Figma page(s) or frame(s)
+that best exhibits associated component designs. Note that the parameter URLs
+are internal-only and point to designs that exist in the main Garden Figma
+branch. Therefore, `parameters` may need to be added as a follow-on for new
+components.
+
+### Story
+
+A `Story` type component is the best way to develop demo code for non-trivial
+Garden components. Doing so removes complexity of coding in MDX, placing
+component rendering and application of story args into TypeScript where code can
+be properly maintained. In its basic form, a `Story` looks like this:
+
+```tsx
+import React from 'react';
+import { Story } from '@storybook/react';\
+import { Component, IComponentProps } from '@zendeskgarden/react-package';
+
+interface IArgs extends IComponentProps {
+  /* Subcomponent and "Story" arg definitions go here */
+}
+
+export const ComponentStory: Story<IArgs> = args => (
+  <Component {...args} />
+);
+```
+
+Take time to ensure Story `args` and `argTypes` are defined well. Often you will
+find that the majority of demo development time is spent on `args` and
+`argTypes` and the `Story` itself essentially falls out of the goal to
+demonstrate component flexibility in isolation.
+
+## Naming
+
+- Follow the existing Storybook naming hierarcy:
+  `Packages/{{PackageName}}/{{ComponentName}}`
+- The first story in every package demo should be a `#readme.stories.mdx` file
+  that renders the package's README.md. Prefix with a hashtag to ensure this story
+  is ordered first.
+- Subsequent component story files are named `componentName.stories.mdx`
+- Use Garden's standard prop naming for story args (i.e. isXxx/hasXxx for
+  boolean args), but use `argTypes` to rename with subcomponent notation. For
+  example:
+- All `Story` TSX files are placed under a `demo/stories` directory. Additionally:
+  - Move all boilerplate data constants (i.e. control data for a component's
+    `children`) into `demo/stories/data.ts`
+  - Capture all types and interfaces specific to story data under
+    `demo/stories/types.ts`
+- If a package has patterns (see section below), tuck all pattern demo code
+  under a `demo/~patterns` directory. Prefix with a tilde to ensure these
+  stories are ordered last.
```

Comment:
I would have liked to see this sooner in the document as well, possibly under the same `### File structure` section as mentioned in the previous comment.